cloudflare.go

```diff
@@ -464,13 +481,41 @@ func (worker *CloudflareWorker) UpdateIPLists() error {
 		}
 	}
 
-	for action, state := range newIPSetByAction {
-		if reflect.DeepEqual(worker.CFStateByAction[action].IPListState.IPSet, state) || len(state) == 0 {
+	for action, set := range newIPSetByAction {
+		if reflect.DeepEqual(worker.CFStateByAction[action].IPListState.IPSet, set) {
 			log.Info("no changes to IP rules ")
 			continue
 		}
+		if len(set) == 0 {
+			// The ReplaceIPListItemsAsync method doesn't allow to empty the list.
+			// Hence we only add one mock IP and later delete it.
+			set["10.0.0.1"] = struct{}{}
+			defer func(action string) {
```

Comment:
I think we want to comment on why the defer : I guess it's because we later in the function iterate on the list to do the replace, and defer avoids an extra if/check at the end to perform this ?

Reply:
Fixed via https://github.com/crowdsecurity/cs-cloudflare-bouncer/pull/56/commits/8b56fccf8f992071713099b8d387df061530fcee